src/gatsby/models/blog/onCreateMarkdownContentNode.js

```diff
@@ -16,7 +19,13 @@ async function createMarkdownBlogNode(api, { parentNode, createChildNode }) {
     pictureComment
   } = frontmatter
   const { name, relativePath } = parentNode
-
+  let gitDateTime
+  if (process.env.CI) {
+    const { stdout } = await exec(
+      `git log -1 --pretty=format:%aI ${fileAbsolutePath}`
+    )
+    gitDateTime = stdout
```

Comment:
could please describe how exactly do we use this? do we still send all the records to algolia (including this date) or we compare it with something and do not send certain records?

Reply:
We are currently using [`gatsby-plugin-algolia`](https://github.com/algolia/gatsby-plugin-algolia/blob/master/gatsby-node.js) for indexing. 

If we have partial indexing enabled, then what it does is
- first gets the current indexed data from the Algolia with only the fields we specified (for eg: `modified`, `slug`)
- compares it with the freshly generated data and filters the new ones, ones whose fields have changed and removed ones.
- then it generates the data from the filtered and sends only those for indexing.

And for this particular `modified` field we are using the last commit timestamp for each post, which I assumed will remain the same until a new commit is made.